tests/test_peer_storage.nim

```diff
@@ -36,72 +47,53 @@ suite "Peer Storage":
 
     # Test insert and retrieve
 
-    require storage.put(peer.peerId, stored, connectedness, disconn).isOk
+    require storage.put(stored).isOk
 
     var responseCount = 0
 
-    # Fetched variables from callback
-    var resPeerId: PeerId
+    # Fetched variable from callback
     var resStoredInfo: RemotePeerInfo
-    var resConnectedness: Connectedness
-    var resDisconnect: int64
 
-    proc data(peerId: PeerID, storedInfo: RemotePeerInfo,
-              connectedness: Connectedness, disconnectTime: int64) {.raises: [Defect].} =
+    proc data(storedInfo: RemotePeerInfo) =
       responseCount += 1
-
-      # Note: cannot use `check` within `{.raises: [Defect].}` block
-      # @TODO: /Nim/lib/pure/unittest.nim(577, 16) Error: can raise an unlisted exception: Exception
-      # These flags are checked outside this block.
-      resPeerId = peerId
       resStoredInfo = storedInfo
-      resConnectedness = connectedness
-      resDisconnect = disconnectTime
 
     let res = storage.getAll(data)
 
     check:
       res.isErr == false
       responseCount == 1
-      resPeerId == peer.peerId
       resStoredInfo.peerId == peer.peerId
       resStoredInfo.addrs == @[peerLoc]
       resStoredInfo.protocols == @[peerProto]
       resStoredInfo.publicKey == peerKey.getPublicKey().tryGet()
-      # TODO: For compatibility, we don't store connectedness and disconnectTime
-      #resStoredInfo.connectedness == connectedness
-      #resStoredInfo.disconnectTime == disconn
-      resConnectedness == Connectedness.CanConnect
-      resDisconnect == disconn
+      resStoredInfo.connectedness == connectedness
+      resStoredInfo.disconnectTime == disconn
+    
+    if resStoredInfo.enr.isSome():
```

Comment:
Shall we enforce the condition that the `enr` was properly stored?

Maybe something like:
```suggestion
    assert resStoredInfo.enr.isSome(), "The ENR info wasn't properly stored"
```

Reply:
Yes, better.